### PR TITLE
TMDM-14749 Throw error in server log when modify an item in PROVISIONING

### PR DESCRIPTION
**Pull request by hw-zhu**
https://jira.talendforge.org/browse/TMDM-14749
**What is the current behavior?** (You should also link to an open issue here)
The behavior is while inserting data into a table without transactions, that caused the insert statements don't be stored to DB.

**What is the new behavior?**
Add begin transaction during business operation, that new data will get inserted into the table as it got rolled back when an error occurred and we have achieved the atomic by using try/catch block.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java [hunk 1]
```diff
@@ -46,10 +47,17 @@ public DocumentSaver createSaver() {
...
             @Override
             public void save(SaverSession session, DocumentSaverContext context) {
+                final XmlServer xmlServer = Util.getXmlServerCtrlLocal();
                 try {
-                    final XmlServer xmlServer = Util.getXmlServerCtrlLocal();
+                    xmlServer.start(dataCluster);
                     xmlServer.putDocumentFromString(record, null, dataCluster);
+                    xmlServer.commit(dataCluster);
                 } catch (Exception e) {
+                    try {
```
Comment (zbcui) [inline org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:56]: Should we log the Exception e & XtentisException e1 ?

File: org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java [hunk 2]
```diff
@@ -46,10 +47,17 @@ public DocumentSaver createSaver() {
...
+                        xmlServer.rollback(dataCluster);
+                    } catch (XtentisException e1) {
+                        throw new RuntimeException(e1);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:59]: No log here?

File: org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java [hunk 3]
```diff
@@ -46,10 +47,17 @@ public DocumentSaver createSaver() {
...
-                    final XmlServer xmlServer = Util.getXmlServerCtrlLocal();
+                    xmlServer.start(dataCluster);
                     xmlServer.putDocumentFromString(record, null, dataCluster);
+                    xmlServer.commit(dataCluster);
                 } catch (Exception e) {
+                    try {
+                        xmlServer.rollback(dataCluster);
+                    } catch (XtentisException e1) {
+                        throw new RuntimeException("An error occurred when rollback all the statements inside a transaction.", e1);
+                    }
```
Comment (htyin) [inline org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:60]: We always use 'Error occurred...'  without `An`

Comment (lpw82gf) [reply org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:60]: `Error occurred while rolling back a transaction.`

File: org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java (hunk 1 above)
Comment (lpw82gf) [reply org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:56]: After checking other implementations of DocumentSaverContext, on this level only throw exceptions, and the log will on higher levels. We can keep the same.

File: org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java (hunk 3 above)
Comment (hw-zhu) [reply org.talend.mdm.core/src/com/amalto/core/save/context/DirectWriteContext.java:60]: done

**Pull request merged**